From 52f86bb507f4abff3ef9044dbc6d5f0d742dc2de Mon Sep 17 00:00:00 2001
From: kashike <kashike@vq.lc>
Date: Sat, 14 Jan 2017 06:53:06 -0800
Subject: [PATCH] Add source method to TranslatableComponent

---
 .../magnet/api/text/translation/TranslatableComponent.java    | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/main/java/net/avicus/magnet/api/text/translation/TranslatableComponent.java b/src/main/java/net/avicus/magnet/api/text/translation/TranslatableComponent.java
index 9bf08435..825fdc19 100644
--- a/src/main/java/net/avicus/magnet/api/text/translation/TranslatableComponent.java
+++ b/src/main/java/net/avicus/magnet/api/text/translation/TranslatableComponent.java
@@ -1,6 +1,7 @@
 package net.avicus.magnet.api.text.translation;
 
 import net.md_5.bungee.api.chat.BaseComponent;
+import org.bukkit.command.CommandSender;
 
 import java.util.Locale;
 
@@ -16,4 +17,14 @@ public interface TranslatableComponent {
      * @return the translated component
      */
     BaseComponent translate(Locale locale);
+
+    /**
+     * Translate this component into a {@link BaseComponent} in the specified {@link Locale}.
+     *
+     * @param viewer the viewer of the component
+     * @return the translated component
+     */
+    default BaseComponent translate(CommandSender viewer) {
+        return this.translate(viewer.getLocale());
+    }
 }
-- 
2.11.0

