From 4a4598f2f7051d9e824a9c89d11fc22d2cca317f Mon Sep 17 00:00:00 2001
From: Austin Mayes <austin@avicus.net>
Date: Fri, 7 Jul 2017 19:48:46 -0500
Subject: [PATCH] Expose watchers in PacketPlayOutEntityMetadata

---
 .../net/minecraft/server/PacketPlayOutEntityMetadata.java | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/main/java/net/minecraft/server/PacketPlayOutEntityMetadata.java b/src/main/java/net/minecraft/server/PacketPlayOutEntityMetadata.java
index c5be70f8..043ce662 100644
--- a/src/main/java/net/minecraft/server/PacketPlayOutEntityMetadata.java
+++ b/src/main/java/net/minecraft/server/PacketPlayOutEntityMetadata.java
@@ -5,8 +5,8 @@ import java.util.List;
 
 public class PacketPlayOutEntityMetadata implements Packet<PacketListenerPlayOut> {
 
-    private int a;
-    private List<DataWatcher.WatchableObject> b;
+    public int a;
+    public List<DataWatcher.WatchableObject> b;
 
     public PacketPlayOutEntityMetadata() {}
 
@@ -33,8 +33,4 @@ public class PacketPlayOutEntityMetadata implements Packet<PacketListenerPlayOut
     public void a(PacketListenerPlayOut packetlistenerplayout) {
         packetlistenerplayout.a(this);
     }
-
-    public void a(PacketListener packetlistener) {
-        this.a((PacketListenerPlayOut) packetlistener);
-    }
 }
-- 
2.23.0

